ouroboros-consensus/src/Ouroboros/Consensus/Util/EarlyExit.hs

```diff
@@ -183,6 +183,19 @@ instance (MonadMask m, MonadAsync m, MonadCatch (STM m))
   cancel        a  = lift $ cancel     (withEarlyExit a)
   cancelWith    a  = lift . cancelWith (withEarlyExit a)
 
+  -- Is this a good definition?
+  -- Is doing concurrency in `WithEarlyExit` a good idea?
+  -- What does it mean to "exit early" when there are a bunch of threads
+  -- spawning all over the place?
+  -- Seems these concurrency-related instances are given because IOLike is
+  -- needed by `forkBlockProudction` in `NodeKernel`. But really, the "early
+  -- exit" part does not fork any threads, it is only used to define the
+  -- thing that runs in the spawned thread, so morally these instances should
+  -- not be needed and indeed should not be defined.
```

Comment:
@edsko some comments from Alex? What's your take on this?

Reply:
I don't necessarily disagree with Alex here but I also don't think it's particularly high priority. Much the same reasoning could be applied to `IO` itself. We don't try to limit what you can do in `IO` and we don't try to limit what we can do in `IOLike`. Bigger fish to fry at the moment.

Reply:
Ok, I will remove the comment, and if you wish you can add an issue.

Reply:
In fact, I just realized that we can do one better. Opened https://github.com/input-output-hk/ouroboros-network/issues/2058 .